### PR TITLE
Remove legacy functions used for SteadyState

### PR DESCRIPTION
**Pull request by PhilipVinc**
Those things where still around from netket 2.5 and I had never removed them because one function was still used in the gradient of Lindblad squared.

I now moved the body of the only used function there, and removed the old code that is not used anywhere.

Lastly, this also switches to a new logic inside of linblad, that i had never turned on for some reason, which uses nkjax.vjp instead of nkjax.grad. This makes the density matrix code work with non-holomorphic density matrix ansatze

### CODE REVIEW

**Review by gcarleo — APPROVED** [review]
beautiful (if tests pass!)
